.github/workflows/integration-tests.yaml

```
@@ -0,0 +1,35 @@
+name: "Integration tests"
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+  pull-requests: read
+jobs:
+
+  integration-tests:
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -y # fix broken repo cache
+          sudo apt install -y make golang-1.20 python3-pip
+          go install gotest.tools/gotestsum@latest
+          go get -u github.com/proglottis/gpgme
+
+      - name: Deploy DB with helm and kind
+        run: make deploy-db-helm
```

```suggestion
      - name: Setup all dependencies
        uses: ./.github/actions/setup-dependencies
        with:
          setup_podman4: yes
```

I refactored it in another patch.